<?xml version="1.0" ?>
<launch>

    <arg name="map_path" default="/media/tier4/ExtremeSSD2TB/t4b/kashiwanoha"/>
    <arg name="t4b_launch_path" default="$(find scenario_api)/launch/scenario.launch"/>
    <arg name="scenario_path" default=""/>
    <arg name="log_output_dir" default=""/>
    <arg name="scenario_id" default=""/>
    <arg name="scenario_runner_output" default="screen"/>
    <arg name="use_sim_time" default="false"/>
    <param name="/use_sim_time" value="$(arg use_sim_time)"/>

    <node pkg="scenario_runner" type="scenario_runner_node" name="scenario_runner_node" output="$(arg scenario_runner_output)">
        <param name="scenario_id" value="$(arg scenario_id)"/>
        <param name="scenario_path" value="$(arg scenario_path)"/>
        <param name="log_output_path" value="$(arg log_output_dir)/$(arg scenario_id).json"/>
        <remap from="input/pointcloud" to="/sensing/lidar/no_ground/pointcloud" />
        <remap from="input/vectormap" to="/map/vector_map" />
        <remap from="input/route" to="/planning/mission_planning/route" />
        <remap from="input/autoware_state" to="/autoware/state" />
        <remap from="input/vehicle_twist" to="/vehicle/status/twist" />
        <remap from="input/signal_command" to="/vehicle/status/turn_signal" />
        <remap from="output/start_point" to="/initialpose" />
        <remap from="output/initial_velocity" to="/initialtwist" />
        <remap from="output/goal_point" to="/planning/mission_planning/goal" />
        <remap from="output/check_point" to="/planning/mission_planning/checkpoint" />
        <remap from="output/autoware_engage" to="/autoware/engage" />
        <remap from="output/simulator_engage" to="/vehicle/engage" />
        <remap from="output/npc_simulator_engage" to="/simulation/npc_simulator/engage" />
        <remap from="output/limit_velocity" to="/planning/scenario_planning/max_velocity" />
        <remap from="output/object_info" to="/simulation/npc_simulator/object_info" />
        <remap from="output/traffic_detection_result" to="/perception/traffic_light_recognition/traffic_light_states" />
        <remap from="output/lane_change_permission" to="/planning/scenario_planning/lane_driving/lane_change_approval" />
        <remap from="rosparam/add_simulator_noise" to="/simple_planning_simulator/add_measurement_noise" /> <!-- rosparam: add simulator noise or not -->
        <remap from="rosparam/simulator_pos_noise" to="/simple_planning_simulator/pos_noise_stddev" /> <!-- rosparam: std of simulator pos noise -->
        <remap fromTODO="rosparam/max_velocity" to="/planning/scenario_planning/motion_velocity_optimizer/max_velocity" /> <!-- rosparam: max velocity -->
        <param name="camera_frame_id" value="camera5/camera_optical_link" /> <!-- base link of front camera-->
    </node>

    <include file="$(arg t4b_launch_path)">
        <arg name="map_path" default="$(arg map_path)" />
    </include>

    <node pkg="rosbag" type="record" name="rosbag_recorder" output="screen" respawn="false" respawn_delay="0" args="record --lz4 -a -O $(arg log_output_dir)/$(arg scenario_id).bag">
    </node>
</launch>